sunpy/instr/aia/response.py

```diff
@@ -0,0 +1,462 @@
+                           'filtersincludemesh', 'meshtrans', 'usecontam', 'contamthick',
+                           'usephottoelec', 'elecperev', 'usephottodn', 'elecperdn', 'useerror',
+                           'fp_filter', 'ent_filter', 'primary', 'secondary', 'ccd', 'contam',
+                           'cross_area']
+
+        # wavelength center list
+        # TODO: want 6 EUV channels and 2 UV channels --- needs updated
+        self.wavelength_centers = [94, 131, 171, 193, 211, 304, 335]
+
+        #           Notes: I needed a way to define self.dataframe locally inside the aia_inst_genx_to_dataframe
+        #           and afterwards use it globally. This is what i have found, but am open to other suggestions.
+
+        #           I could import os and recursivly search the .genx directory for the files needed rather than the user defining one file. Then, the variable would change to path_to_genx_dir
+
+
+        # keyword arguements
+        path_to_genx_file = kwargs.get('path_to_genx_file', '')
+        save_outfile = kwargs.get('save_outfile', False)
+        csv = kwargs.get('csv', '')
+
+
+        # Load in the filename with the call of the function to initialize channel properties
+        if path_to_genx_file and save_outfile:
+            self.aia_inst_genx_to_dataframe(path_to_genx_file, save_outfile)
+        elif path_to_genx_file:
+            self.aia_inst_genx_to_dataframe(path_to_genx_file)
+        elif csv:
+            self.dataframe = pd.DataFrame.from_csv(csv)
+        else:
+            print('Implement keyword path_to_genx_file or csv to load properties.')
+
+        wavelength = kwargs.get('wavelength', '')
+
+        # assert type(wavelength) == str
+        # wavelength range is the same for all channels, but can be loaded per channel, reflectance is only defined if wavelength is defined for quick plotting
+        if wavelength:
+            if wavelength == 'all':
+                wrange = []
+                reflect = []
+                for i in self.wavelength_centers:
+                    wrange.append(self.dataframe[str(i)]['wave'])
+                    reflect.append(self.dataframe[str(i)]['primary'] * self.dataframe[str(i)]['secondary'])
+                self.wavelength_range = wrange
+                self.reflectance = reflect
+            else:
+                # add units
+                self.wavelength_range = self.dataframe[wavelength]['wave']
+                self.reflectance = (self.dataframe[wavelength]['primary']) * (
+                    self.dataframe[str(wavelength)]['secondary'])
+        else:
+            self.wavelength_range = self.dataframe['94']['wave']
+
+
+            # Notes: I'm trying to define self.stuffs as cleanly as possible - Any input appreciated as this is still new to me
+            #          I saw that def __repr__(self): can be used to define self.things but I'm not sure on how..?
```

Comment:
__repr__() defines what is displayed in the console when you print an instance of your class - I'm not sure what you mean by using it to define attributes except in the sense that you can do so anywhere in the class definition in the way you've been doing it.